add outputbinding kitex

liu-song opened the pull request:
# Description
add outputbinding kitex 
<!--
Please explain the changes you've made.
-->

## Issue reference
https://github.com/dapr/dapr/pull/6093
https://github.com/dapr/components-contrib/pull/2563

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

liu-song commented on cmd/daprd/components/binding_kitex.go:
```diff
@@ -0,0 +1,26 @@
+//go:build !(linux && arm)
+// +build !linux !arm
```
note: Using the Go build tag to skip testing is not compiling on the Linux ARM platform.

mukundansundar reviewed (changes requested):
@liu-song Is there possibility of the underlying libraries adding support for Linux Arm architecture?

liu-song commented:
Thank you for your response.@mukundansundar

Currently, it can be compiled on a Linux arm64 architecture, but compilation on an arm32 system fails. The build tag I added was to prevent compilation on arm32 systems.

To fully support compilation on arm32 systems, it is necessary to make the three libraries that Kitex depends on compatible with the arm32 platform. https://github.com/choleraehyq/pid , https://github.com/cloudwego/netpoll , https://github.com/bytedance/gopkg  However, I do not have the ability to do this compatibility work at the moment.

I will try to contact the maintainers of these three libraries to see if they can make them compatible with arm32. However, the necessity of this compatibility may have some uncertainty in terms of time, so for now, I added the go build tag to address the issue.

berndverst commented:
@liu-song the @dapr/maintainers-dapr have now agreed that we will add build flag and will exclude this comment for ARM32 builds of Dapr. I'll update your PR.

liu-song commented:
That's great, thank you!

berndverst commented:
@mukundansundar do you know what's going on with the test failures?

Also have you decided whether the increases binary size due to this component (but also updating components-contrib in general) is acceptable?

mukundansundar commented:
> @mukundansundar do you know what's going on with the test failures?
> 
> Also have you decided whether the increases binary size due to this component (but also updating components-contrib in general) is acceptable?

the tests are flaky. There is an issue open to tackle it.

liu-song commented:
@berndverst ,hi
It seems that the failing test function is not related to this PR. May I know if we can proceed to merge this PR now? 
If yes, I will update https://github.com/dapr/docs/issues/3266.

berndverst commented:
Once https://github.com/dapr/dapr/pull/6290 is merged and working, we will merge this PR.

Please note: This component will be considered `beta`. We are likely changing the stable definition such that we will not allow components to become stable which increase our binary size too much. At the moment this means that Kitex can never become stable due to the increase in binary size.

liu-song commented:
> Once #6290 is merged and working, we will merge this PR.
> 
> Please note: This component will be considered `beta`. We are likely changing the stable definition such that we will not allow components to become stable which increase our binary size too much. At the moment this means that Kitex can never become stable due to the increase in binary size.

Agreed, I have updated the build tag for all_components.

yaron2 commented:
@liu-song please resolve the conflict and we can merge this.

liu-song commented:
Alright, I have resolved this conflict.
